### PR TITLE
Adjust handling for downstream MST devices

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: libfwupdplugin/fu-dpaux-device.c [hunk 1]
```diff
@@ -105,16 +84,18 @@ fu_dpaux_device_setup(FuDevice *device, GError **error)
 	FuDpauxDevicePrivate *priv = GET_PRIVATE(self);
 	guint8 buf[FU_STRUCT_DPAUX_DPCD_SIZE] = {0x0};
 	g_autoptr(GByteArray) st = NULL;
+	const gchar *name;
 
+	name = fu_device_get_name(device);
```
Comment (hughsie) [inline libfwupdplugin/fu-dpaux-device.c:89]: maybe just loose the `name` local var?

File: plugins/synaptics-mst/fu-synaptics-mst-device.c [hunk 2]
```diff
@@ -1627,9 +1628,18 @@ fu_synaptics_mst_device_setup(FuDevice *device, GError **error)
...
 					   (FuDeviceLockerFunc)fu_synaptics_mst_device_disable_rc,
-					   error);
-	if (locker == NULL)
+					   &error_local);
+	if (locker == NULL) {
+		if (g_strcmp0(fu_device_get_name(device), "DPMST") == 0) {
+			g_set_error_literal(error,
+					    G_IO_ERROR,
+					    G_IO_ERROR_NOT_SUPPORTED,
+					    "downstream endpoint not supported");
```
Comment (hughsie) [inline plugins/synaptics-mst/fu-synaptics-mst-device.c:1637]: do we want to return with an error, or just skip the locker completely for DPMST devices?

Comment (superm1) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c:1637]: I don't think we want to be continuing setup unless we could communicate with the device.

File: libfwupdplugin/fu-dpaux-device.c (hunk 1 above)
Comment (superm1) [reply libfwupdplugin/fu-dpaux-device.c:89]: Sure

File: plugins/synaptics-mst/fu-synaptics-mst-device.c (hunk 2 above)
Comment (hughsie) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c:1637]: Ohh I figured it was just the RC that was failing -- and the other stuff worked without it. I've got a horrible feeling we might need to do the RC manually on the root node to be able to upgrade the DPMST device. In the meantime, this seems okay to me.

Comment (superm1) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c:1637]: I don't think there are **really** two Synaptics MST hubs connected in this bug.  But once we have a setup with two connected cascaded we can see what's **really** needed.

**Review by hughsie — APPROVED** [review]

**Pull request merged**